YAXLib/Deserialization.cs

```diff
@@ -862,8 +886,18 @@ private bool TrySetValueForEmptyElement(object obj, MemberWrapper member, Type m
         return null;
     }
 
+    private object? TryDataItemListDirect(Type collType, IList dataItems)
```

Comment:
Would be more clean to apply the "TryGet" pattern, that typically involves a method that attempts to retrieve a out value or object and returns a boolean indicating whether the retrieval was successful or not.